server/routes/getExposureOverTime.ts

```diff
@@ -0,0 +1,145 @@
+import Sequelize from 'sequelize';
+import { Exposure, BlockNumber, AccountId, IndividualExposure } from '@polkadot/types/interfaces';
+import { Request, Response, NextFunction } from 'express';
+import { Errors } from './getOffences';
+const Op = Sequelize.Op;
+
+interface IEventData {
+    stash: AccountId;
+    exposure: Exposure;
+    block_number: BlockNumber;
+}
+
+// eslint-disable-next-line prefer-const
+let validators: { [key: string]: any[] } = {};
+
+// COMMON
+const getStakeOverTime = async (models, req: Request, res: Response, next: NextFunction) => {
+  const { chain, stash } = req.query;
+  let { startDate, endDate } = req.query;
+
+  if (!chain) return next(new Error(Errors.ChainIdNotFound));
+  const chainInfo = await models.Chain.findOne({ where: { id: chain } });
+  if (!chainInfo) return next(new Error(Errors.InvalidChain));
+
+  // if start and end date isn't given, we set it for a year from now
+  if (typeof startDate === 'undefined' || typeof endDate === 'undefined') {
+    startDate = new Date();
+    startDate.setFullYear(startDate.getFullYear() - 1);
+    startDate = new Date(startDate);
+    endDate = new Date();
+  }
+
+  const where: any = {
+    '$ChainEventType.chain$': chain,
+  };
+
+  // if stash is given
+  if (stash) where['$HistoricalValidatorStatistic.stash$'] = stash;
+
+  if (startDate && endDate) {
+    where.created_at = {
+      [Op.between]: [startDate, endDate]
+    };
+  }
+
+  const stakeOverTime = await models.HistoricalValidatorStatistic.findAll({
+    where,
+    order: [
+      ['created_at', 'ASC']
+    ],
+    attributes: ['stash', 'exposure', 'block'],
+    include: [ { model: models.ChainEventType } ]
```

Comment:
can you describe what adding this model is doing and how you are sure this gets you the data you expect?

Reply:
This was an unnecessary join, I have made the changes in https://github.com/hicommonwealth/commonwealth/pull/606/commits/8283f4f320fa9dbd99307a95a9803296b6415bd4 at **_tmp_furqan-555-556-594_**